### PR TITLE
Merge lf module

### PR DESCRIPTION
**Pull request by owm111**
### Description

Add module `programs.lf`, for managing [lf](https://github.com/gokcehan/lf)'s config.

### Checklist

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all`.
    > There are failing tests (xsession-basic, xsession-keyboard-without-layout, and lieer-service) but they appear to be irrelevant to the new module. Not checking for now for this reason.

- [x] Test cases updated/added. See [example](https://github.com/rycee/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/rycee/home-manager/blob/master/CONTRIBUTING.md#commits) for more information and [recent commit messages](https://github.com/rycee/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [ ] Added myself as module maintainer. See [example](https://github.com/rycee/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [x] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

**Comment by owm111** [comment]
After fixing an error that started appearing after upgrading to 20.03, the other tests specified in the original request are still failing for me. However, all tests pass on Travis CI, so I'm going to assume everything is fine (and the issue is with my machine).

File: modules/programs/lf.nix [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+        default = "";
+        example = ''
+          $mkdir -p ~/.trash
+        '';
+        description = "Custom lfrc lines.";
+      };
+    };
+  };
+
+  config = let
```
Comment (rycee) [inline modules/programs/lf.nix:169]: I think you can remove the empty `let` block.

File: modules/programs/lf.nix [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+      settings = mkOption {
+        type = lfSettingsType;
+        default = { };
+        example = {
+          tabstop = 4;
+          number = true;
+          ratios = "1:1:2";
+        };
+        description = ''
+          An attribute set of lf settings. The attribute names and cooresponding
```
Comment (rycee) [inline modules/programs/lf.nix:68]:
```suggestion
          An attribute set of lf settings. The attribute names and corresponding
```

**Comment by owm111** [comment]
I added a `meta.maintainers` for myself, however I am not listed on nixpkgs/maintainers/maintainers-list.nix. Will this be an issue?

**Comment by rycee** [comment]
Thanks! Rebased to master in 2f2a4396c617f843e8e9adf74687e73d592a80ab.

**Pull request closed without merging**